internal/agent/collector/vsphere.go

```diff
@@ -633,6 +644,12 @@ func getDatastores(hosts *[]vspheremodel.Host, collector *vsphere.Collector) []a
 	res := []apiplanner.Datastore{}
 	for _, ds := range *datastores {
 		dsVendor, dsModel, dsProtocol := findDataStoreInfo(*hosts, ds.BackingDevicesNames)
+		hostName := findDatastoreHost(*hosts, ds.ID)
```

Comment:
A suggestion:
Maybe better to create map before the for loop and than use this map
This will improve the performance, avoid going through all datastores each time
What do you think?